File: java/spring-boot-webmvc.g8/src/main/g8/src/test/java/$package$/ApplicationTest.java
package $package$;

import static org.assertj.core.api.Assertions.assertThat;

import $package$.note.Note;
import $package$.note.NoteRepository;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

@SpringBootTest
class ApplicationTest {

    @Autowired
    private NoteRepository notes;

    @Test
    void contextLoadsAndPersists() {
        Note saved = notes.save(new Note("hello"));
        assertThat(saved.getId()).isNotNull();
        assertThat(notes.findById(saved.getId())).hasValueSatisfying(n ->
                assertThat(n.getText()).isEqualTo("hello"));
    }
}

File: kotlin/spring-boot/webmvc-security-actuator-jpa-h2.g8/src/main/g8/$if(simple.truthy)$test/src$else$src/test/kotlin$endif$/$package$/ApplicationTest.kt
package $package$

import $package$.note.Note
import $package$.note.NoteRepository
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest
class ApplicationTest {

    @Autowired
    lateinit var notes: NoteRepository

    @Test
    fun contextLoadsAndPersists() {
        val saved = notes.save(Note("hello"))
        assertThat(saved.id).isNotNull()
        assertThat(notes.findById(saved.id!!).get().text).isEqualTo("hello")
    }
}
